File: src-py/src/subsystems/sensors.py

from wpilib.command.subsystem import Subsystem

from wpilib import RobotDrive

from values import ids

from wpilib.analoginput import AnalogInput
from wpilib.ultrasonic import Ultrasonic

class Sensors(Subsystem):

	def __init__(self):
		super().__init__('Sensors')
		self.ultrasonic0 = Ultrasonic(ids.ULTRASONIC0, ids.ULTRASONIC0)
		#self.ultrasonic0 = AnalogInput(ids.ULTRASONIC0, ids.ULTRASONIC0)
		#self.ultrasonic0.setOversampleBits(4)
		#self.ultrasonic0.setAverageBits(2)

	def getDistanceIn(self):
		return self.ultrasonic0.getRangeInches()

File: project/src/team3966/robot/subsystems/Sensors.java
/*
 * L&N STEMpunks c 2017
 *
 * WOPR-JR.
 *
 * Full repo: github.com/ln-stempunks/WOPR-JR
 *
 * Full licensing here: programming.lnstempunks.org/licensing
 *
 * GPLv3
 */
package team3966.robot.subsystems;

import com.kauailabs.navx.frc.AHRS;
import edu.wpi.first.wpilibj.I2C;
import edu.wpi.first.wpilibj.command.Subsystem;
import edu.wpi.first.wpilibj.SPI;

import team3966.robot.values.IDs;
import team3966.robot.hardware.DistanceSensor;
import team3966.robot.hardware.Lidar;
import team3966.robot.hardware.PDPManager;

public class Sensors extends Subsystem {

    public DistanceSensor ultrasonic;
    public Lidar lidar;
    public AHRS navX;
    public PDPManager pdp;

    public Sensors() {
        ultrasonic = new DistanceSensor(IDs.ultrasonic_0);
        navX = new AHRS(SPI.Port.kMXP);
        pdp = new PDPManager();
        lidar = new Lidar();
    }
    public double getLidarDifference () {
        double difference = lidar.getInches() / lidar.getDistance();
        return difference;
    }
    protected void initDefaultCommand() {

    }

}
